core-processor/src/ext.rs

```diff
@@ -319,6 +319,134 @@ impl GrowHandler for LazyGrowHandler {
     }
 }
 
+enum Change {
+    MarkUsed(ReservationId),
+}
+
+/// Used to atomically update `Ext` which prevents some errors
+/// when data was updated but operation failed.
+///
+/// Copies some counters into itself and performs operations on them and
+/// incrementally builds list of changes.
+///
+/// Changes are applied after operation is completed
+struct ExtChanges {
+    gas_counter: GasCounter,
+    gas_allowance_counter: GasAllowanceCounter,
+    value_counter: ValueCounter,
+    outgoing_glassless: u64,
```

Comment:
gas-less (wout gas applied)
```suggestion
    outgoing_gasless: u64,
```